pennylane/math/single_dispatch.py

```diff
@@ -295,7 +295,10 @@ def _round_tf(tensor, decimals=0):
 
 def _ndim_tf(tensor):
     try:
-        return _i("tf").experimental.numpy.ndim(tensor)
+        ndim = _i("tf").experimental.numpy.ndim(tensor)
+        if ndim is None:
+            raise AttributeError

```

Comment:
```suggestion
            return len(tensor.shape)
```

Since an attribute error indicates to return `len(tensor.shape)` anyway.